anker/src/wormhole.rs

```diff
@@ -0,0 +1,283 @@
+use std::str::FromStr;
+
+use crate::{error::AnkerError, token::MicroUst};
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use hex::FromHexError;
+use serde::Serialize;
+use solana_program::{
+    entrypoint::ProgramResult,
+    instruction::{AccountMeta, Instruction},
+    msg,
+    pubkey::Pubkey,
+};
+
+/// Wormhole's Terra chain id.
+pub const WORMHOLE_CHAIN_ID_TERRA: u16 = 3;
+
+/// Position of the native transfer code at the Wormhole project:
+/// https://github.com/certusone/wormhole/blob/05425a96df6e5841f05e7be5e7f4c45be01985a6/solana/modules/token_bridge/program/src/lib.rs
+const WORMHOLE_NATIVE_TRANSFER_CODE: u8 = 5;
+
+#[repr(C)]
+#[derive(
+    Clone, Default, Debug, BorshSerialize, BorshDeserialize, BorshSchema, Eq, PartialEq, Serialize,
+)]
+pub struct ForeignAddress([u8; 32]);
+
+impl FromStr for ForeignAddress {
+    type Err = FromHexError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut foreign_address = [0; 32];
+        // Allow for array starting with 0x
+        let array_vec = if s.len() >= 2 && &s[..2] == "0x" {
+            hex::decode(&s[2..])
+        } else {
+            hex::decode(s)
+        }?;
```

Comment:
We can make this a bit nicer with [`strip_prefix`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix):

```rust
match s.strip_prefix("0x") {
    Some(suffix) => hex::decode(suffix),
    None => hex::decode(s),
}?;
```
or even
```rust
hex::decode(s.strip_prefix("0x").unwrap_or(s))?;
```

Reply:
Actually, don’t bother with this, I think we shouldn’t convert to/from hexadecimal, we should use Bech32 which is used by Terra. (No need to do that in this PR, it’s already big enough; we can change this later.)

Reply:
I see there's something in #471 already. Will leave it.